applications/Melodot.md

```diff
@@ -0,0 +1,238 @@
+3. **Consensus-related extensions**
+
+**`excutive_das`** pallet: An extension of the **`frame-excutive`** pallet for scheduling block execution and validation related to DAS
+
+**`system_das`** pallet: An extension of the **`frame-system`** pallet, adding new block header generation, actual data generation, and validation.
+
+4. **Sampling Core**
+
+The core crate that actually performs sampling, which several clients in the system depend on. This includes data availability sampling, obtaining and propagating data from the DHT network, and managing data through rocksdb.
+
+5. **Light Client**
+
+A light client that connects to the network and performs sampling based on the block state, including network implementation, distributed generation, data storage, and actual sampling.
+
+6. **`Melo-PoSpace`**
+
+A pallet for assigning distributed generation columns to farmers, verifying farmers' challenge chunk, and issuing rewards.
+
+7. **Farmer Client**
+
+Connects to the network, obtains raw data, and performs distributed generation; obtains challenges from the chain and searches for solutions to claim rewards.
+
+### **Non-Goals**
+
+The goal of the first phase is to minimally implement a usable system and will not fully implement the details described in the whitepaper. The main differences are:
+
+**PoSpace**
+
+In the first phase, we will only implement a preliminary version of PoSpace, not including the complete PoSpace process. [Subspace has done an excellent exploration in this regard](https://subspacelabs.notion.site/Subspace-Dilithium-Consensus-Specification-v2-3-274a730b53eb4c93a8d879b90de532ce). In the next phase of development, we should be able to reuse much of their work.
+
+**Complete Distributed Generation**
+
+This phase does not include complete distributed generation. Users still need to submit actual data transactions, so farmers and validators do not need to perform the first phase of sampling.
+
+## **Ecosystem Fit**
+
+### Similar Projects
+
+There are currently several data availability layer projects, including Ethereum Danksharding, Celestia, Avail, and Eigen DA. Our main difference from them is the introduction of farmers, which solves many tricky problems faced by the data availability layer. Unlike Danksharding, we decouple an independent data availability layer, which is the same principle as Polkadot not supporting smart contracts. Celestia uses a Merkle encoding pattern, requiring fraud proofs and additional assumptions, which we avoid. Avail's data layout uses a 1.5D scheme, which is unacceptable in terms of sampling efficiency, as detailed in the Melodot white paper. Eigen DA is an Ethereum re-collateralization layer implementation of a data availability scheme, with limited public information available, [but it should be in the form of a DA committee](https://ethresear.ch/t/unbundling-pbs-towards-protocol-enforced-proposer-commitments-pepc/13879), which has its applicable scenarios, but is not the same as Melodot.
```

Comment:
It also reminds me of the old proof-of-capacity project [Signum (formerly Burstcoin)](https://signum.network/), combined with Chia's proof-of-space. Can you explain who would be the primary users of this project once it is complete?

Reply:
Yes, there are various implementations of proof-of-space / proof-of-capacity. The difference lies in the fact that Melodot utilizes them to store availability data,  so the primary users of Melodot are rollup applications.